Godeps/_workspace/src/k8s.io/kubernetes/pkg/registry/generic/etcd/etcd.go

```diff
@@ -459,6 +457,11 @@ func (e *Etcd) DeleteCollection(ctx api.Context, options *api.DeleteOptions, lis
 			return nil, err
 		}
 	}
+
+	// Try to delete the directory, which will only succeed if the directory is empty.
+	// We can ignore errors since the cost of an empty directory is relatively small.
+	e.Storage.DeleteDir(ctx, e.KeyRootFunc(ctx));
```

Comment:
can we not call this when list options has provided a field selector or label selector?

delete collection lets you scope the set of items to delete....